nilearn/maskers/tests/test_nifti_labels_masker.py

```diff
@@ -319,227 +354,262 @@ def test_nifti_labels_masker_reduction_strategies():
+    labels = Nifti1Image(labels_data, affine_eye)
 
     # What NiftiLabelsMasker should return for each reduction strategy?
-    expected_results = {
-        "mean": np.mean(test_values),
-        "median": np.median(test_values),
-        "sum": np.sum(test_values),
-        "minimum": np.min(test_values),
-        "maximum": np.max(test_values),
-        "standard_deviation": np.std(test_values),
-        "variance": np.var(test_values),
-    }
-
-    for strategy, expected_result in expected_results.items():
-        masker = NiftiLabelsMasker(labels, strategy=strategy)
-        # Here passing [img] within a list because it's a 3D object.
-        result = masker.fit_transform([img]).squeeze()
-        assert result == expected_result
+    expected_result = function(test_values)
 
-    with pytest.raises(ValueError, match="Invalid strategy 'TESTRAISE'"):
-        NiftiLabelsMasker(labels, strategy="TESTRAISE")
+    masker = NiftiLabelsMasker(labels, strategy=strategy)
+    # Here passing [img] within a list because it's a 3D object.
+    result = masker.fit_transform([img]).squeeze()
+
+    assert result == expected_result
 
     default_masker = NiftiLabelsMasker(labels)
+
     assert default_masker.strategy == "mean"
 
 
-def test_nifti_labels_masker_resampling(tmp_path):
-    """Test resampling in NiftiLabelsMasker."""
-    n_regions = 9
-    length = 3
+def test_nifti_labels_masker_reduction_strategies_error(affine_eye):
+    """Tests NiftiLabelsMasker invalid strategy."""
+    labels_data = np.array([[[0, 0, 0, 0, 0], [1, 1, 1, 1, 1]]], dtype=np.int8)
+
+    labels = Nifti1Image(labels_data, affine_eye)
+
+    with pytest.raises(ValueError, match="Invalid strategy 'TESTRAISE'"):
+        NiftiLabelsMasker(labels, strategy="TESTRAISE")
 
-    shape1 = (10, 11, 12, length)
-    affine = np.eye(4)
 
+def test_nifti_labels_masker_resampling_errors(
+    affine_eye, shape_3d_default, n_regions
+):
+    """Test errors of resampling in NiftiLabelsMasker."""
+    labels_img = generate_labeled_regions(
+        shape=shape_3d_default,
+        n_regions=n_regions,
+        affine=affine_eye,
+    )
+
+    with pytest.raises(ValueError):
+        NiftiLabelsMasker(labels_img, resampling_target="mask")
+
+    with pytest.raises(ValueError):
+        NiftiLabelsMasker(
+            labels_img,
+            resampling_target="invalid",
+        )
+
+
+def test_nifti_labels_masker_resampling_to_data(
+    tmp_path, affine_eye, n_regions, length
+):
+    """Test resampling to data in NiftiLabelsMasker."""
     # mask
-    shape2 = (16, 17, 18, length)
+    shape2 = (8, 9, 10, length)
+    # maps
+    shape3 = (16, 18, 20)
+
+    _, mask22_img = generate_random_img(
+        shape2,
+        affine=affine_eye,
+    )
+
+    labels33_img = generate_labeled_regions(
+        shape3, n_regions, affine=affine_eye
+    )
+
+    # Test with data and atlas of different shape:
+    # the atlas should be resampled to the data
```

Comment:
IIUC, both atlas _and mask_ would be resampled?

Reply:
From the doc:

For example, if resampling_target is “data”, the atlas is resampled to the shape of the data if needed. 

If it is “labels” then mask_img and images provided to fit() are resampled to the shape and affine of maps_img. 

“None” means no resampling: if shapes and affines do not match, a ValueError is raised.

Reply:
When the target is `labels` there is a check in the test to make sure the mask is resampled.

I don't see a test for the `None` case though so actually worth adding it.

Reply:
I see! So mask_img does not need to be resampled to data because it is applied to the labels_img (not the data). As mentioned [here](https://github.com/nilearn/nilearn/blob/a950195e83d6f9af64fa3166369427b8a58d1066/nilearn/maskers/nifti_labels_masker.py#L73):
"Mask to apply to regions before extracting signals."

Reply:
Thanks!